File: domain/src/main/java/de/x4fyr/timeline/domain/ExternalElement.java
package de.x4fyr.timeline.domain;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NonNull;

import java.time.Duration;
import java.time.LocalDateTime;
import java.util.UUID;

/**
 * @author Benedikt Volkmer
 *         Created on 11/4/16.
 */
@Data
@EqualsAndHashCode(callSuper = true)
public class ExternalElement extends ScheduledElement {

    @NonNull
    private UUID externalUuid;
    @NonNull
    private boolean scheduled = false;


    public ExternalElement(@NonNull UUID uuid, LocalDateTime start, Duration duration, String
            title, String notes) {
        super(start, duration, title, notes);
        this.externalUuid = uuid;
    }

}

File: domain/src/main/kotlin/timeline/domain/elements/ExternalElement.kt
package timeline.domain.elements

import utils.date.DateTime
import utils.date.Duration

/**
 * Element coming from an external source like a calendar.
 * @author Benedikt Volkmer
 *         Created on 11/4/16.
 */
data class ExternalElement(val id: Long? = null,
                           var title: String,
                           var notes: String,
                           var start: DateTime,
                           var duration: Duration,
                           internal val scheduledId: Long? = null)

